src/models/users.js

```diff
@@ -9,22 +6,65 @@ import {
 
 // Supported authentication methods.
 const authMethods = {
-  BASIC: 'basic'
+  BASIC: ({ username, password }) => {
+    // For now we only support admin authentication.
+    if (username !== 'admin' || password !== config.get('adminPass')) {
+      return Promise.reject(new Error(UNAUTHORIZED));
+    }
+
+    return Promise.resolve({
+      id: 'admin',
+      scope: 'admin'
+    });
+  },
+
+  AUTH_PROVIDER: (data) => {
+    return Promise.resolve({
+      id: data,
+      scope: 'user'
+    });
+  },
+};
+
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('Users', {
+    opaqueId: { type: DataTypes.STRING(256) },
+    provider: { type: DataTypes.STRING(32) },
+  }, {
+    indexes: [{
+      unique: true,
+      fields: ['opaqueId', 'provider', 'ClientKey']
+    }]
+  });
+
+  User.authenticate = (method, data) => {
+    if (!authMethods[method]) {
+      return Promise.reject(new Error(UNSUPPORTED_AUTH_METHOD));
+    }
+
+    return authMethods[method](data)
+      .then(userData => {
+        if (userData.scope !== 'user') {
+          return userData;
+        }
+
+        return User.findOrCreate({
+          attributes: [],
+          where: userData.id, // this contains all user attributes
+        })
+          .then(() => userData);
```

Comment:
Ok. Thanks for the explanation.

Regarding the indentation, it's not a big deal, but rather that introducing a new one, try to follow the existing style, please.

```javascript
return User.findOrCreate({
...
}).then(...
```
or even
```javascript
return User.findOrCreate({
...
})
.then(...
```